src/main/java/org/junitpioneer/jupiter/StdIntercept.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Intercept communication on the standard input/output,
+ * to verify behaviour. This should be used with
+ * {@link org.junitpioneer.jupiter.StdIOExtension.StdOut}
+ * to intercept the standard output ({@code System.in}) and
+ * {@link org.junitpioneer.jupiter.StdIOExtension.StdIn}
+ * to intercept the standard input ({@code System.out}).
+ */
+@Retention(RetentionPolicy.RUNTIME)
+public @interface StdIntercept {
```

Comment:
I see you've took my idea of calling it `Interceptor` - but why cutting the last two character then?

Reply:
I try to name everything in my code so it looks like written language when in context. For example here, `StdInterceptor` looks better, but if you read the test it becomes:
`@StdIntercept StdIn in` - which is saying (to me) *intercept StdIn with the variable **in***.
To be honest, I originally wanted to rename this annotation just `@Intercept` but that looks too general (if that makes sense).

Reply:
The annotation is unnecessary for resolving StdOut and StdIn could use a method level annotation instead for getting its parameters, for example: `@StdInSource`. Would that be simpler?

Reply:
> The annotation is unnecessary for resolving StdOut and StdIn could use a method level annotation instead for getting its parameters, for example: `@StdInSource`. Would that be simpler?

The missing need for `StdOut` should be documented in the final `adoc` !

Reply:
This was changed and documented.